File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        lowerCaseS = s.lower()
        trimmedStr = "".join(char for char in lowerCaseS if char.isalnum())

        left, right = 0, len(trimmedStr) - 1

        while left < len(trimmedStr):
            if trimmedStr[left] != trimmedStr[right]:
                return False
            
            left += 1
            right -= 1
        
        return True

File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    const trimmedString = s.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
    let [left, right] = [0, trimmedString.length - 1];
    while (left < trimmedString.length - 1) {
        if (trimmedString[left] !== trimmedString[right]){
            return false
        }
        left++;
        right--;
    }

    return true
};